torch/csrc/distributed/rpc/rref_context.h

```diff
@@ -195,17 +196,24 @@ class TORCH_API RRefContext {
       RRefId::Hash>
       forks_;
 
-  // The follow two maps keep UserRRefs alive by holding a intrusive_ptr to the
+  std::condition_variable pendingReduceCV_;
```

Comment:
need comments on this CV. BTW, why there is a `Reduce` in the name?

Reply:
I am using mussel memory from that 2 cond vars on producer-consumer queue, one is for item added, the other for item removed. 

Since we are waiting on both sizes to be 0, I think only need to notify `pendingUsers_.erase(iter)` and `pendingChildren_.erase(iter)`. `pendingUsers_.empalce(**)` and `pendingChildren_.emplace(**)` don't need to notify this CV.